website/docs/Use-Cases/agent_chat.md

```diff
@@ -113,7 +113,9 @@ def exchange_rate(base_currency: CurrencySymbol, quote_currency: CurrencySymbol)
     else:
         raise ValueError(f"Unknown currencies {base_currency}, {quote_currency}")
 
-
+# NOTE: for Azure OpenAI, API version up to 2023-10-01-preview
+# uses the old function call style. So you need to set `api_style="function"`:
```

Comment:
https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/function-calling?tabs=python
This doc says 2023-12-01-preview supports tool call.
But in my own test, 2023-08-01 also supports it. And 2023-12-01-preview also supports function call.

Reply:
So, then it is likely that the underlying API was actually updated. But we can't rely on unspecified behavior of the earlier API versions. So, I think we should still need this backward compatibility fix and the changes in #1229.

Reply:
I think the right instruction is: For Azure OpenAI API, use 2023-12-01-preview or later versions. "functions" will be deprecated in future.

Reply:
Sometime the azure open ai endpoint might have been provisioned by someone other than the developer building the app. So the developer can’t change the api version.

Reply:
According to the website, other versions are unsupported for function call.

Reply:
Right